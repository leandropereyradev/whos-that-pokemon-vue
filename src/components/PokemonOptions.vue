<template>
  <div class="options-container">
    <ul>
      <li v-for="pokemon in pokemons" :key="pokemon.id">{{ capitalize(pokemon.name) }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'Pokemon Options',
  props: {
    pokemons: {
      type: Array,
      required: true
    }
  },
  methods: {
    capitalize(name) {
      return name.charAt(0).toUpperCase() + name.slice(1)
    }
  }
}
</script>

<style scoped>
ul {
  list-style-type: none;
}

li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>