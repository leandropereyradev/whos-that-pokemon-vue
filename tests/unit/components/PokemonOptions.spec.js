import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import { mockPokemons } from "../mocks/pokemons.mock";

describe("PokemonOptions component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons: mockPokemons,
      },
    });
  });

  test("Should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("Should display all four options correctly", () => {
    const buttonTags = wrapper.findAll("button");

    expect(buttonTags.length).toBe(4);

    expect(buttonTags[0].text()).toBe("Bulbasaur");

    expect(buttonTags[1].text()).toBe("Ivysaur");

    expect(buttonTags[2].text()).toBe("Venusaur");

    expect(buttonTags[3].text()).toBe("Charmander");
  });

  test('Should output "selected-pokemon" with its parameters on click', () => {
    const buttonTags = wrapper.findAll("button");

    buttonTags[0].trigger("click");
    buttonTags[1].trigger("click");
    buttonTags[2].trigger("click");
    buttonTags[3].trigger("click");

    expect(wrapper.emitted("selected-pokemon").length).toBe(4);

    expect(wrapper.emitted("selected-pokemon")[0]).toEqual([1]);

    expect(wrapper.emitted("selected-pokemon")[1]).toEqual([2]);

    expect(wrapper.emitted("selected-pokemon")[2]).toEqual([3]);

    expect(wrapper.emitted("selected-pokemon")[3]).toEqual([4]);
  });
});
