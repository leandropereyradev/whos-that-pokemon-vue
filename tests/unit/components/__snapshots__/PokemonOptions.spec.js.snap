// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Should match the snapshot 1`] = `<div class="options-container"><button class="btnOption btn animation isActive">Bulbasaur</button><button class="btnOption btn animation isActive">Ivysaur</button><button class="btnOption btn animation isActive">Venusaur</button><button class="btnOption btn animation isActive">Charmander</button></div>`;
